torch/onnx/symbolic_opset9.py

```diff
@@ -1966,6 +1966,24 @@ def hardsigmoid(g, self):
     # See https://pytorch.org/docs/stable/generated/torch.nn.Hardsigmoid.html
     return g.op("HardSigmoid", self, alpha_f=1 / 6)
 
+@parse_args("v")
+def tanhshrink(g, self):
+    return g.op("Sub", self, tanh(g, self))
+
+@parse_args("v", "f")
+def hardshrink(g, self, lambd):
+    lambd_op = g.op("Constant", value_t=torch.FloatTensor([lambd]))
+    cond = logical_or(g, gt(g, self, lambd_op), lt(g, self, neg(g, lambd_op)))
+    return g.op("Where", cond, self, g.op("Constant", value_t=torch.FloatTensor([0])))
```

Comment:
Can where handle operands of different dtypes?

Reply:
Hardshrink and softshrink seem to accept only float and double as input types, so I do not think this should be an issue